src/warp.jl

```diff
@@ -0,0 +1,38 @@
+# None of these types are provided by this package, so the following line seems unclean
+#     @inline Base.getindex(A::AbstractExtrapolation, v::StaticVector) = A[convert(Tuple, v)...]
+# furthermore it would be ambiguous with getindex(::Extrapolation, xs...) after https://github.com/tlycken/Interpolations.jl/pull/141
+@inline _getindex(A, v::StaticVector) = A[convert(Tuple, v)...]
+
+warp(img::AbstractArray, args...) = warp(interpolate(img, BSpline(Linear()), OnGrid()), args...)
+
+function warp{T<:Colorant,S}(::Type{T}, img::AbstractArray{S}, args...)
+    TCol = typeof(convert(T,one(S)))
+    TNorm = eltype(TCol)
+    apad, pad = Interpolations.prefilter(TNorm, TCol, img, typeof(BSpline(Linear())), typeof(OnGrid()))
+    itp = Interpolations.BSplineInterpolation(TNorm, apad, BSpline(Linear()), OnGrid(), pad)
+    warp(itp, args...)
+end
+
+@compat const FloatLike{T<:AbstractFloat} = Union{T,Gray{T}} # Why not AbstractGray ?
+@compat const FloatColorant{T<:AbstractFloat} = Colorant{T}
+
+# The default values used by extrapolation for off-domain points
+@inline _default_fill{T<:FloatLike}(::Type{T}) = convert(T, NaN)
+@inline _default_fill{T<:FloatColorant}(::Type{T}) = nan(T)
+@inline _default_fill{T}(::Type{T}) = zero(T)
+
+warp{T}(img::AbstractInterpolation{T}, tform, fill=_default_fill(T)) = warp(extrapolate(img, fill), tform)
```

Comment:
I added a third parameter to pass the extrapolation scheme / fill value. This allows things like

```julia
warp(img, tfm, Interpolations.Flat())
warp(img, tfm, 1)
```